### PR TITLE
[FileVersion] Update site version policy commands

### PR DESCRIPTION
**Pull request by msjennywu**
Before creating a pull request, make sure that you have read the contribution file located at

https://github.com/pnp/powerShell/blob/dev/CONTRIBUTING.md

## Type ##
- [X] Bug Fix
- [ ] New Feature
- [ ] Sample

## What is in this Pull Request ? ##
1. Add new cmdlet Set-PnPSiteVersionPolicy, remove the change from Set-PnPSite

2. Properties updated comparing with new Set-PnPSiteVersionPolicy with Set-PnPSite
MinorVersions -> MajorWithMinorVersions
ApplyForNewLibs -> ApplyToNewDocumentLibraries
ApplyForExistingLibs -> ApplyToExistingDocumentLibraries
InheritTenantVPForNewDocLibs -> InheritFromTenant
CancelVPForExistingLibs -> CancelForExistingDocumentLibraries

3. Update PnP cmdlet name
Get-PnPSiteSetVersionPolicyProgress -> Get-PnPSiteVersionPolicyProgress

4. Update progress properties name
ListsProcessedInTotal -> LibrariesProcessedInTotal
ListsFailedInTotal -> LibrariesFailedInTotal
EnableAutoTrim -> EnableAutomaticMode

5. Check if "No Site Level Policy Set" by "MajorVersionLimit"
VersionPolicies is a null value if no site level policy is set. The PnP powershell needs to be re-login or re-open to get the refreshed value after getting the null value and resetting as not-null value.
MajorVersionLimit is updated as int value in Sharepoint now, return value is -1 for "No Site Level Policy Set", so not above issue.

6. Add message when run set version policy on site for new and existing document libraries
7. Add info "for new document libraries" in description of result from cmdlet Get-PnPSiteVersionPolicy

Test scenarios for new cmdlet:
----Apply to both new and existing doc libs
--AutoExpiration
Set-PnPSiteVersionPolicy -EnableAutoExpirationVersionTrim $true
Set-PnPSiteVersionPolicy -EnableAutoExpirationVersionTrim $true -ApplyToNewDocumentLibraries -ApplyToExistingDocumentLibraries

--ExpireAfter
Set-PnPSiteVersionPolicy -EnableAutoExpirationVersionTrim $false -MajorVersions 100 -MajorWithMinorVersions 10 -ExpireVersionsAfterDays 200
Set-PnPSiteVersionPolicy -EnableAutoExpirationVersionTrim $false -MajorVersions 101 -MajorWithMinorVersions 11 -ExpireVersionsAfterDays 201 -ApplyToNewDocumentLibraries -ApplyToExistingDocumentLibraries

--NoExpiration
Set-PnPSiteVersionPolicy -EnableAutoExpirationVersionTrim $false -MajorVersions 300 -MajorWithMinorVersions 20 -ExpireVersionsAfterDays 0
Set-PnPSiteVersionPolicy -EnableAutoExpirationVersionTrim $false -MajorVersions 300 -MajorWithMinorVersions 11 -ExpireVersionsAfterDays 0 -ApplyToNewDocumentLibraries -ApplyToExistingDocumentLibraries

--Apply to new doc libs only
--AutoExpiration
Set-PnPSiteVersionPolicy -EnableAutoExpirationVersionTrim $true -ApplyToNewDocumentLibraries

--ExpireAfter
Set-PnPSiteVersionPolicy -EnableAutoExpirationVersionTrim $false -MajorVersions 100 -ExpireVersionsAfterDays 200 -ApplyToNewDocumentLibraries

--NoExpiration
Set-PnPSiteVersionPolicy -EnableAutoExpirationVersionTrim $false -MajorVersions 300 -ExpireVersionsAfterDays 0 -ApplyToNewDocumentLibraries

--Clear
Set-PnPSiteVersionPolicy -InheritFromTenant

----Apply to existing doc libs only
--AutoExpiration
Set-PnPSiteVersionPolicy -EnableAutoExpirationVersionTrim $true -ApplyToExistingDocumentLibraries

--ExpireAfter
Set-PnPSiteVersionPolicy -EnableAutoExpirationVersionTrim $false -MajorVersions 100 -MajorWithMinorVersions 10 -ExpireVersionsAfterDays 50 -ApplyToExistingDocumentLibraries

--NoExpiration
Set-PnPSiteVersionPolicy -EnableAutoExpirationVersionTrim $false -MajorVersions 300 -MajorWithMinorVersions 11 -ExpireVersionsAfterDays 0 -ApplyToExistingDocumentLibraries

--Cancel request for existing doc libs
Set-PnPSiteVersionPolicy -CancelForExistingDocumentLibraries

Some example test results:
image

### CODE REVIEW

**Comment by msjennywu** [comment]
@KoenZomers Could you please help to review this PR? Thanks!

**Comment by gautamdsheth** [comment]
hi @msjennywu , plurals are not allowed as per PowerShell convention, it needs to be singular . 

Also, `Get-PnPSiteSetVPProgressExistingLibs` this contains `Get` and `Set` both which doesn't make sense, it should be a clear `Get` since we are fetching the progress.

**Comment by msjennywu** [comment]
> hi @msjennywu , plurals are not allowed as per PowerShell convention, it needs to be singular .
> 
> Also, `Get-PnPSiteSetVPProgressExistingLibs` this contains `Get` and `Set` both which doesn't make sense, it should be a clear `Get` since we are fetching the progress.

@gautamdsheth  Thanks for the review!
The command `Get-PnPSiteSetVPProgressExistingLibs` is for getting "set version policy for existing libs on the site", that's why there is "Set" in the command, but it is not the command verb. The plurals, do you mean the "Libs"? there are multiple libs on the site, so it is plural. Please let me know if you still have the concern. Thanks a log!

**Comment by msjennywu** [comment]
@KoenZomers  I have fixed your concerns. Could you please review the PR again? Thanks a bunch!

**Comment by msjennywu** [comment]
@gautamdsheth I have fixed your concerns in new iteration, could you please review it again? Thanks a bunch!

File: documentation/Set-PnPSiteVersionPolicy.md [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+schema: 2.0.0
+applicable: SharePoint Online
+external help file: PnP.PowerShell.dll-Help.xml
+online version: https://pnp.github.io/powershell/cmdlets/Set-PnPSiteVersionPolicy.html
+---
+ 
+# Set-PnPSiteVersionPolicy
+
+## SYNOPSIS
+Sets site collection properties.
```
Comment (KoenZomers) [inline documentation/Set-PnPSiteVersionPolicy.md:13]: Could you be a little more descriptive what this cmdlet can be used for?

**Pull request merged**